### DISCUSSION TITLE
0.9.7 - Mostly bug fixes with some new options

### DISCUSSION THREAD

**Discussion by Timthreetwelve** [Timthreetwelve/GetMyIP#52] [Announcements]
### This update includes new options and fixes some bugs 🐛🐜🐞 

Apart from bug fixes and translation updates, I expect the pace of updates to Get My IP to slow down as I turn my attention to projects that have been neglected for too long. When I return to Get My IP in 2024, I will be upgrading it to .NET 8, which was released earlier this month and is the next LTS version of .NET.

#### Changes:
- Fixed: Settings will now be saved every time a user navigates away from the Settings page. Settings were saved at shutdown, but this can be an issue with a long running application.
- Fixed: If a newer version of Get My IP is found when checking for an updated release from the About page, clicking OK in the resulting message box will now close Get My IP after navigating to the GitHub Releases page.
- Fixed: Only one Instance of Get My IP will be allowed to run at a time. This fixes the scenario described in Issue #51.  Logging to a permanent log file is exempted from the only one instance rule.
- Fixed: Logging to a permanent log file was broken in d1e9f05. It now works correctly.
- Fixed: The list of translation contributors on the about page is now aligned correctly. This didn't cause any harm, it just bugged me.
- New: Added [Free IP API](https://freeipapi.com) as an IP information provider. It's good to have options.
- New: The list of IP information providers used by Get My IP is now available at the bottom of the about page. 
- New: An option to show external IP type (IPv4 or IPv6) in the tray icon tooltip has been added.

#### Thanks:
Last week was the Thanksgiving 🦃 holiday in the US. **I'm thankful** for **everyone** that has contributed language translations and to those that are keeping them updated! 🙏🏻

If you wish to contribute a translation, check out the [Contribute a Translation](https://github.com/Timthreetwelve/GetMyIP/wiki/Contribute-a-Translation) topic in the Wiki. Your help would be awesome.

**Full Changelog**: https://github.com/Timthreetwelve/GetMyIP/compare/v0.9.6...v0.9.7

#### Checksums :
|File|SHA256 hash|
|----|---------------|
|Installer (exe)| `7334a685aae4c4d2f581df38d6a51c5b8d94956de6bd89cc76e83b0f8484724d` |
|Portable (zip)| `a8f92633a235ee2233552af7066289f5a49e45b51c3c92d30900853eb4211318` |

<hr /><em>This discussion was created from the release <a href='https://github.com/Timthreetwelve/GetMyIP/releases/tag/'>0.9.7 - Mostly bug fixes with some new options</a>.</em>

**Comment by logounet** [comment]
Hello,

Here is the French translation (based upon latest en-US) for "GetMyIP_0.9.7.0"

No errors/problems on validation

![Presse-papier02](https://github.com/Timthreetwelve/GetMyIP/assets/15344607/4c85c176-dad2-4560-a0c4-0d5ce67dc395)


See attached file

[Strings.fr-FR.xaml.zip](https://github.com/Timthreetwelve/GetMyIP/files/13477067/Strings.fr-FR.xaml.zip)

Ps: 2 threads cannot be translated, see picture

![Presse-papier01](https://github.com/Timthreetwelve/GetMyIP/assets/15344607/1e22f7d9-8b9f-469a-9580-c65ff8b2ddda)



Best regards

**Comment by Timthreetwelve** [comment]
@logounet 

Thanks so much for the translation! ❤️

Thanks for pointing out the two missing strings. I added them moments ago. 

``` xml
    <sys:String x:Key="SettingsItem_StartWithWindows">Start with Windows</sys:String>
    <sys:String x:Key="SettingsItem_StartMinimized">Start Minimized</sys:String>
```

You could update the xaml file and attach it again or reply to this discussion with the two strings. Either way is fine. Once that is done I will open an issue for the new translation, just so that I can keep track.

Thanks again,
Tim

**Comment by logounet** [comment]
Here with the two strings

[Strings.fr-FR.xaml.zip](https://github.com/Timthreetwelve/GetMyIP/files/13477435/Strings.fr-FR.xaml.zip)

Best regards

**Comment by Timthreetwelve** [comment]
Thanks so much!

Tim

**Comment by logounet** [comment]
Hello,

Any small modification in my translation:

"Afficher Quitter dans la barre de navigation" instead of  "Afficher la sortie dans la barre de navigation"

[Strings.fr-FR.xaml.zip](https://github.com/Timthreetwelve/GetMyIP/files/13673002/Strings.fr-FR.xaml.zip)


Best regards
